arch/arm/src/armv7-m/arm_initialstate.c

```diff
@@ -65,8 +66,9 @@ void up_initial_state(struct tcb_s *tcb)
     {
       tcb->stack_alloc_ptr = (void *)(g_idle_topstack -
                                       CONFIG_IDLETHREAD_STACKSIZE);
-      tcb->stack_base_ptr   = tcb->stack_alloc_ptr;
-      tcb->adj_stack_size  = CONFIG_IDLETHREAD_STACKSIZE;
+      tcb->stack_base_ptr  = tcb->stack_alloc_ptr;
+      tcb->adj_stack_size  = CONFIG_IDLETHREAD_STACKSIZE -
+                             sizeof(struct task_info_s);
```

Comment:
1. Shouldn't the task info be initialized to zero?

    memset(tcb->stack_alloc_ptr, 0, sizeof(sizeof(struct task_info_s));

2. This does not work in SMP mode.  There are multiple IDLE tasks and they do not all have pid == 0

Same comments on all similar changes.

Reply:
The better fix is call up_stack_frame/task_setup_info at:
https://github.com/apache/incubator-nuttx/blob/master/sched/init/nx_smpstart.c#L133
https://github.com/apache/incubator-nuttx/blob/master/sched/init/nx_start.c#L519
Just like other normal thread.